src/utils/v2/credentialsUtils.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtServiceClientCredentials, ISubscriptionContext } from '@microsoft/vscode-azext-dev';
+import * as vscode from 'vscode';
+import { AzureSubscription } from '../../../api/src/index';
+import * as vscodeAccount from '../../tree/azure/VSCodeAuthentication';
+import { localize } from '../../utils/localize';
+
+/**
+ * Converts a VS Code authentication session to an Azure Track 1 & 2 compatible compatible credential.
+ */
+export function createCredential(getSession: (scopes?: string[]) => vscode.ProviderResult<vscode.AuthenticationSession>): AzExtServiceClientCredentials {
```

Comment:
How does this differ from `src/tree/azure/VSCodeAuthentication.ts`?

Reply:
![image](https://user-images.githubusercontent.com/5290572/227132560-6cd81095-9e35-4e66-a78a-0563f9310cbc.png)